src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyEolSwitch.java

```diff
@@ -0,0 +1,57 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.leftcurly;
+/**
+  * This test-input is intended to be checked using following configuration:
+  *
+  * option = EOL
+  *
+  */
+public class InputLeftCurlyEolSwitch {
+
+    public void doStuff() {
+        int x = 1;
+        switch (x) {
+            case 0:
+            { // warn
+                break;
+            }
+            case (1+0):
+            { // warn
+                break;
+            }
+            case 2: {
+                break;
+            }
+            default:
+            { // warn
+                break;
+            }
+            case 3:
+            case 4:
+                x++;
+                { // OK, standalone block
```

Comment:
what about if `{}` is first, followed by `x++`? it should be treated as a standalone block.
please add it to all options.

Reply:
@romani please confirm you agree, there should be no validation unless brace spans the entire case.

Reply:
>what about if {} is first, followed by x++? it should be treated as a standalone block.
please add it to all options.

Sounds reasonable to me. What about this case:
```
            case 5:  {
                }
                break;
```
Should it be a standalone block or a case block?

Reply:
Done. After some research, I decided that it makes sense to force users rewrite such blocks as
```
            case 5:  {
                    break;
                }
```
So, there should be only one child in `case` or `default`, and the child must be a `SLIST`.

Reply:
there might be numerous use cases where users play with extra blocks to get code more readable.
Be careful with extra demands from user, as you do not know user's intend.

> So, there should be only one child in case or default, and the child must be a SLIST

this is not a business of this this Check to demand this.
It will create one move wave of issues to not validate this.

>```             case 5:  {
                }
                break;
```

this should not be a problem `eol`. Fact that smth in out of some visibility scope is not business of this Check. Check should validate only first block left curly. This Should mentioned in documentation. All other blocks that are under case should be ignored.
There should be another Check that demands one block under case/default.

Reply:
Done. Reverted to 01763be62661482889a63ad4f3aeb7d29e58e352 with some additional test cases.